pkg/controller/machine.go

```diff
@@ -793,6 +855,25 @@ func (c *controller) updateMachineStatus(
 	return clone, nil
 }
 
+// isMachineStatusEqual checks if the status of 2 machines is similar or not.
+func isMachineStatusEqual(s1, s2 v1alpha1.MachineStatus) bool {
+	tolerateTimeDiff := 30 * time.Minute
+	s1Copy, s2Copy := s1.DeepCopy(), s2.DeepCopy()
+	s1Copy.LastOperation.Description, s2Copy.LastOperation.Description = "", ""
+
+	if (s1Copy.LastOperation.LastUpdateTime.Time.After(time.Now().Add(tolerateTimeDiff))) || (s2Copy.LastOperation.LastUpdateTime.Time.After(time.Now().Add(tolerateTimeDiff))) {
```

Comment:
`time.Now().Add(tolerateTimeDiff)` is 30 minutes into the future. Should it be `Sub()` instead?

Reply:
That's a good catch.

Reply:
This is why we need @ggaurav10 's eyes